cmd/analyse_test.go

```diff
@@ -0,0 +1,245 @@
+package cmd_test
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"math"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/damianoneill/nc-hammer/cmd"
+	"github.com/damianoneill/nc-hammer/result"
+	. "github.com/damianoneill/nc-hammer/suite"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"gonum.org/v1/gonum/stat"
+)
+
+// Test variables used to populate []result.NetconfResult used throughout
+var (
+	ts1 = result.NetconfResult{Client: 5, SessionID: 2318, Hostname: "172.26.138.91", Operation: "edit-config", When: 55282, Err: "", Latency: 288}
+	ts2 = result.NetconfResult{Client: 6, SessionID: 859, Hostname: "172.26.138.92", Operation: "get-config", When: 55943, Err: "", Latency: 176}
+	ts3 = result.NetconfResult{Client: 4, SessionID: 601, Hostname: "172.26.138.93", Operation: "get", When: 9840, Err: "", Latency: 3320}
+	ts4 = result.NetconfResult{Client: 4, SessionID: 2322, Hostname: "172.26.138.91", Operation: "get", When: 56967, Err: "", Latency: 420}
+	ts5 = result.NetconfResult{Client: 4, SessionID: 860, Hostname: "172.26.138.92", Operation: "kill-session", When: 0, Err: "kill-session is not a supported operation", Latency: 0}
+)
+
+func TestSortResults(t *testing.T) {
+
+	testSort := func(t *testing.T, unsortedSlice []result.NetconfResult, want []result.NetconfResult) {
+		t.Helper()
+
+		SortResults(unsortedSlice)
+		got := unsortedSlice
+
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("got %v, want %v", got, want)
+		}
+	}
+
+	t.Run("Sort by Hostname", func(t *testing.T) {
+		unsortedSlice := []result.NetconfResult{ts3, ts1, ts2}
+		want := []result.NetconfResult{ts1, ts2, ts3}
+
+		testSort(t, unsortedSlice, want)
+	})
+
+	t.Run("Sort by Operation", func(t *testing.T) {
+		unsortedSlice := []result.NetconfResult{ts3, ts4, ts2, ts5, ts1}
+		want := []result.NetconfResult{ts1, ts4, ts2, ts5, ts3}
+
+		testSort(t, unsortedSlice, want)
+	})
+}
+
+func TestOrderAndExcludeErrValues(t *testing.T) {
+	testResults := []result.NetconfResult{ts1, ts2, ts3, ts4, ts5}
+	testLatencies := make(map[string]map[string][]float64)
+
+	got := OrderAndExcludeErrValues(testResults, testLatencies)
+
+	if got != 1 {
+		t.Errorf("got %v, want %v", 1, got)
+	}
+}
+
+func TestAnalyseResults(t *testing.T) {
+	var latencies = make(map[string]map[string][]float64)
+	var mockCmd *cobra.Command
+	var mockResults = []result.NetconfResult{ts1, ts2, ts3}
+	var mockTs = TestSuite{}
+	mockTs.File = "testdata/emptytestsuite.yml"
+
+	// Capture StdErr
+	var lOut = new(bytes.Buffer)
```

Comment:
careful with this naming, historically c developers named variable by putting the first initial of a type in lowercase, a c developer would read this as long Out.